kotlin-codegen/compiler/src/main/java/com/squareup/moshi/JsonClassCodeGenProcessor.kt

```diff
@@ -127,6 +130,17 @@ class JsonClassCodeGenProcessor : KotlinAbstractProcessor(), KotlinMetadataUtils
       nameResolver.getString(it.name)
     }
 
+    // The compiler might emit methods just so it has a place to put annotations. Find these.
+    val annotatedElements = mutableMapOf<Property, ExecutableElement>()
+    for (enclosedElement in element.enclosedElements) {
+      if (enclosedElement is ExecutableElement) {
+        val property = classData.getPropertyOrNull(enclosedElement)
+        if (property != null) {
```

Comment:
You could write this as

```kotlin
val annotatedElements = element.enclosedElements
    .filterIsInstance<ExecutableElement>()
    .mapNotNull { element -> 
      classData.getPropertyOrNull(element)?.let {  it to element } 
    }
    .associate { it }
```

Reply:
start with asSequence to make it single iteration too

Reply:
The implicit Pair in that is awkward. Procedural code is easy to step through in the debugger!

Reply:
How’s this:
```
    // The compiler might emit methods just so it has a place to put annotations. Find these.
    val annotatedElements = mutableMapOf<Property, ExecutableElement>()
    for (enclosedElement in element.enclosedElements) {
      if (enclosedElement !is ExecutableElement) continue
      val property = classData.getPropertyOrNull(enclosedElement) ?: continue
      annotatedElements[property] = enclosedElement
    }
```